src/com/bzu/softwareconstruction/segments/SegmentPartUrl.java

```diff
@@ -0,0 +1,87 @@
+package com.bzu.softwareconstruction.segments;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.bzu.softwareconstruction.services.multipart.FilePartUrl;
+import com.bzu.softwareconstruction.services.multipart.exceptions.InvalidItemURL;
+
+public class SegmentPartUrl implements FilePartUrl{
+	private static final Logger log = Logger.getLogger(SegmentPartUrl.class.getName());
+
+	private String url;
+	private String contentType;
+	
+	public SegmentPartUrl(String url){
+		if(url == null || url.isEmpty()){
+			throw new IllegalArgumentException("url not must have a value");
+		}
+		this.url = url;
+	}
+
+	public SegmentPartUrl(File file) {
+		url = file.toURI().toString();
+	}
+
+	@Override
+	public InputStream openStream() {
+		try {
+			URL url = new URL(this.url);
+			log.log(Level.INFO, this.url);
+			
+			URLConnection connection = url.openConnection();
+			connection.connect();
+
+			//contentType = connection.getContentType();
+			String contentType = connection.getContentType();
```

Comment:
contentType should also be set into data member as well - you maybe change this line to be:
this.contentType = connection.getContentType();